consensus/core/src/storage/mod.rs

```diff
@@ -0,0 +1,17 @@
+// Copyright (c) Mysten Labs, Inc.
+// SPDX-License-Identifier: Apache-2.0
+
+mod rocksdb;
+
+use crate::{block::VerifiedBlock, commit::Commit, error::ConsensusResult};
+
+/// A common interface for consensus storage.
+pub(crate) trait Store {
+    /// Loads last committed blocks, all uncommitted blocks and last commit from store.
+    fn recover(&self) -> ConsensusResult<(Vec<VerifiedBlock>, Commit)>;
+
+    /// Writes additional uncommitted blocks and commits to store.
```

Comment:
just to clarify this can be used to write either uncomitted blocks OR committed blocks + commit metadata?

Reply:
The idea is to batch writes when proposing a block. All accepted but not persisted blocks, and new consensus commits, will be written to the store. Reworded the comment a bit.